sources/models/src/modeled_types/shared.rs

```diff
@@ -420,3 +419,95 @@ mod test_dns_domain {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// SysctlKey represents a string that is a valid Linux sysctl key; keys must be representable as
+/// filesystem paths, and are generally kept to lowercase_underscored_names separated with '.' or
+/// '/'.
+/// It stores the original string and makes it accessible through standard traits. Its purpose is
+/// input validation, for example validating the kubelet's "clusterDomain" config value.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct SysctlKey {
+    inner: String,
+}
+
+lazy_static! {
+    /// Pattern matching the name of a sysctl key.  Must be representable as a path; we'll go a bit
+    /// further and enforce a basic pattern that would match all known keys, plus some leeway.
+    pub(crate) static ref SYSCTL_KEY: Regex = Regex::new(r"^[a-zA-Z0-9./_-]{1,128}$").unwrap();
+}
+
+impl TryFrom<&str> for SysctlKey {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        ensure!(
+            SYSCTL_KEY.is_match(input),
+            error::InvalidSysctlKey {
+                input: input,
+                msg: format!("must match pattern {}", *SYSCTL_KEY),
+            }
+        );
+        Ok(SysctlKey {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(SysctlKey, "SysctlKey");
+
+#[cfg(test)]
+mod test_sysctl_key {
+    use super::SysctlKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_sysctl_key() {
+        for ok in &[
+            // Longest real one
+            "net/ipv4/conf/enp0s42f3/igmpv3_unsolicited_report_interval",
```

Comment:
Would we want to support mixed dots and slashes within a single key? Let's test for it either way